### PR TITLE
Overhaul build script - KSP-CKAN/CKAN#1273

### PR DESCRIPTION
**Pull request by techman83**
This looks to be doing the correct things. Won't try and validate itself. Exits with failures seemingly when it should.

Thanks to @mgsdk for the dummy install function.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -1,59 +1,159 @@
...
+    KSP_NAME=$KSP_NAME_DEFAULT
+    
+    # Set the version to the requested KSP version if supplied.
+    if [ $# -eq 2 ]
+    then
+        KSP_VERSION=$1
+        KSP_NAME=$2
+    fi
+    
+    # TODO: Manual hack, a better way to handle this kind of identifiers may be needed.
```
Comment (plague006) [inline build.sh]: Maybe add a manual hack for `any` as well? If KSP_VERSION == "any" then KSP_VERSION=$KSP_VERSION_DEFAULT

**Comment by plague006** [comment]
Will this close #551?

**Comment by techman83** [comment]
Should do. Missed that @dazpoet had done some work on this.

File: build.sh [hunk 2]
```diff
@@ -1,59 +1,261 @@
...
+    mkdir dummy_ksp/Ships/@thumbs/VAB
+    mkdir dummy_ksp/Ships/@thumbs/SPH
+    
+    echo "Version $KSP_VERSION" > dummy_ksp/readme.txt
+    
+    # Copy in resources.
+    cp ckan.exe dummy_ksp/ckan.exe
+    
+    # Reset the Mono registry.
+    if [ "$USER" = "jenkins" ]
```
Comment (pjf) [inline build.sh]: I bet this will cause a surprise if we ever end up with a contributor who's username is *actually* jenkins. :)

**Comment by pjf** [comment]
1. This looks fine to me.
2. The worst that happens if we merge this and it goofs up is that we undo the merge. :)

So, guess what I'm doing...

**Pull request merged**